### PR TITLE
add pass enhance for map_matmul_to_mul_pass and flatten2_matmul_fuse_…

### PR DESCRIPTION
**Pull request by MissPenguin**
### PR types

New features
### PR changes

Others
### Describe

add pass enhance for map_matmul_to_mul_pass and flatten2_matmul_fuse_pass

### CODE REVIEW

File: paddle/fluid/framework/ir/map_matmul_to_mul_pass.cc [hunk 1]
```diff
@@ -26,6 +27,95 @@ namespace ir {
...
+      .IsTensor()
+      .End()
+      .AddInput("Y")
+      .IsTensor()
+      .End()
+      .AddOutput("Out")
+      .IsTensor()
+      .End()
+      .AddAttr("alpha")
+      .End()
```
Comment (winter-wang) [inline paddle/fluid/framework/ir/map_matmul_to_mul_pass.cc:44]: 缺条件

File: paddle/fluid/framework/ir/map_matmul_to_mul_pass.cc [hunk 2]
```diff
@@ -26,6 +27,95 @@ namespace ir {
...
+      .AddAttr("transpose_X")
+      .End()
```
Comment (winter-wang) [inline paddle/fluid/framework/ir/map_matmul_to_mul_pass.cc:47]: 同上，matmul转mul的话，transpose_X得是false吧

File: paddle/fluid/framework/ir/map_matmul_to_mul_pass.cc [hunk 3]
```diff
@@ -26,6 +27,95 @@ namespace ir {
...
+      .AddAttr("transpose_Y")
```
Comment (winter-wang) [inline paddle/fluid/framework/ir/map_matmul_to_mul_pass.cc:48]: 同上

File: paddle/fluid/framework/ir/map_matmul_to_mul_pass.cc [hunk 4]
```diff
@@ -26,6 +27,95 @@ namespace ir {
...
+      .IsTensor()
+      .End()
+      .AddOutput("Out")
+      .IsTensor()
+      .End()
+      .AddAttr("x_num_col_dims")
+      .IsNumGE(1)
+      .End()
+      .AddAttr("y_num_col_dims")
+      .IsNumGE(1)
```
Comment (winter-wang) [inline paddle/fluid/framework/ir/map_matmul_to_mul_pass.cc]: y_num_c ol_dims应该强等于1，而不是大于等于1

File: paddle/fluid/framework/ir/map_matmul_to_mul_pass.cc [hunk 5]
```diff
@@ -26,6 +27,95 @@ namespace ir {
...
+      .AddInput("Y")
+      .IsTensor()
+      .End()
+      .AddOutput("Out")
+      .IsTensor()
+      .End()
+      .AddAttr("alpha")
+      .End()
+      .AddAttr("transpose_X")
+      .End()
```
Comment (winter-wang) [inline paddle/fluid/framework/ir/map_matmul_to_mul_pass.cc:87]: 同上，缺条件。后面的都一样，缺乏条件

File: paddle/fluid/framework/ir/map_matmul_to_mul_pass.cc [hunk 6]
```diff
@@ -39,6 +129,11 @@ void MapMatmul2MulPass::ApplyImpl(ir::Graph* graph) const {
   int found_count = 0;
   auto handler = [&](const GraphPatternDetector::subgraph_t& subgraph,
                      Graph* g) {
+    if (!IsCompat(subgraph, g)) {
+      LOG(WARNING) << "Pass in op compat failed.";
+      return;
+    }
+
```
Comment (winter-wang) [inline paddle/fluid/framework/ir/map_matmul_to_mul_pass.cc:144]: 79行需要补充对mul op的兼容性判断

**Review by winter-wang — CHANGES_REQUESTED** [review]

File: paddle/fluid/framework/ir/map_matmul_to_mul_pass.cc (hunk 1 above)
Comment (MissPenguin) [reply paddle/fluid/framework/ir/map_matmul_to_mul_pass.cc:44]: done, thanks

File: paddle/fluid/framework/ir/map_matmul_to_mul_pass.cc (hunk 2 above)
Comment (MissPenguin) [reply paddle/fluid/framework/ir/map_matmul_to_mul_pass.cc:47]: done, thanks

File: paddle/fluid/framework/ir/map_matmul_to_mul_pass.cc (hunk 5 above)
Comment (MissPenguin) [reply paddle/fluid/framework/ir/map_matmul_to_mul_pass.cc:87]: done, thanks

File: paddle/fluid/framework/ir/map_matmul_to_mul_pass.cc (hunk 6 above)
Comment (MissPenguin) [reply paddle/fluid/framework/ir/map_matmul_to_mul_pass.cc:144]: done, thanks

File: paddle/fluid/framework/ir/map_matmul_to_mul_pass.cc [hunk 7]
```diff
@@ -26,6 +27,103 @@ namespace ir {
...
+      .IsTensor()
+      .End()
+      .AddInput("Y")
+      .IsTensor()
+      .End()
+      .AddOutput("Out")
+      .IsTensor()
+      .End()
+      .AddAttr("x_num_col_dims")
+      .IsNumGE(1)
```
Comment (winter-wang) [inline paddle/fluid/framework/ir/map_matmul_to_mul_pass.cc:63]: x_num_col_dims的约束应为  IsIntIn({1,2})

**Review by winter-wang — APPROVED** [review]
LGTM

**Review by Shixiaowei02 — APPROVED** [review]

**Pull request merged**